src/input.service.ts

```diff
@@ -453,170 +450,34 @@ export class InputService {
-   * necessary. It schedules itself again provided that there's still
-   * a connected gamepad somewhere.
-   */
-  private pollGamepad(now: number) {
-    const rawpads = Array.from(navigator.getGamepads()).filter(pad => !!pad); // refreshes all checked-out gamepads
-
-    for (let i = 0; i < rawpads.length; i += 1) {
-      const gamepad = this.gamepads[rawpads[i].id];
-      if (!gamepad) {
-        continue;
-      }
-      gamepad.pad = rawpads[i];
-
-      if (!gamepad.isConnected()) {
-        delete this.gamepads[rawpads[i].id];
-        continue;
-      }
-
-      if (this.keyboardVisible) {
-        continue;
-      }
-
-      if (gamepad.left(now)) {
-        const ev = this.focus.createArcEvent(Direction.LEFT);
-        this.handleDirection(ev);
-        this.onLeft.emit(ev);
-      }
-      if (gamepad.right(now)) {
-        const ev = this.focus.createArcEvent(Direction.RIGHT);
-        this.handleDirection(ev);
-        this.onRight.emit(ev);
-      }
-      if (gamepad.down(now)) {
-        const ev = this.focus.createArcEvent(Direction.DOWN);
-        this.handleDirection(ev);
-        this.onDown.emit(ev);
-      }
-      if (gamepad.up(now)) {
-        const ev = this.focus.createArcEvent(Direction.UP);
-        this.handleDirection(ev);
-        this.onUp.emit(ev);
-      }
-      if (gamepad.tabLeft(now)) {
-        const ev = this.focus.createArcEvent(Direction.TABLEFT);
-        this.onLeftTab.emit(ev);
-      }
-      if (gamepad.tabRight(now)) {
-        const ev = this.focus.createArcEvent(Direction.TABRIGHT);
-        this.onRightTab.emit(ev);
-      }
-      if (gamepad.tabDown(now)) {
-        const ev = this.focus.createArcEvent(Direction.TABDOWN);
-        this.onRightTrigger.emit(ev);
-      }
-      if (gamepad.tabUp(now)) {
-        const ev = this.focus.createArcEvent(Direction.TABUP);
-        this.onLeftTrigger.emit(ev);
-      }
-      if (gamepad.view(now)) {
-        const ev = this.focus.createArcEvent(Direction.VIEW);
-        this.onView.emit(ev);
-      }
-      if (gamepad.menu(now)) {
-        const ev = this.focus.createArcEvent(Direction.MENU);
-        this.onMenu.emit(ev);
-      }
-      if (gamepad.submit(now)) {
-        const ev = this.focus.createArcEvent(Direction.SUBMIT);
-        this.handleDirection(ev);
-        this.onAPressed.emit(ev);
-      }
-      if (gamepad.back(now)) {
-        const ev = this.focus.createArcEvent(Direction.BACK);
-        this.handleDirection(ev);
-        this.onBPressed.emit(ev);
-      }
-      if (gamepad.x(now)) {
-        const ev = this.focus.createArcEvent(Direction.X);
-        this.onXPressed.emit(ev);
-      }
-      if (gamepad.y(now)) {
-        const ev = this.focus.createArcEvent(Direction.Y);
-        this.onYPressed.emit(ev);
-      }
-    }
-
-    if (Object.keys(this.gamepads).length > 0) {
-      this.scheduleGamepadPoll();
-    } else {
-      this.pollRaf = null;
-    }
-  }
-
```

Comment:
Please do not remove this, this is necessary for desktop gamepad support / support for gamepad on platforms which don't support keyboard emulation